lesson_14/homework_14.py

```diff
@@ -29,4 +30,67 @@
     print("Користувачі різні")
 
 Написати на це все тести
-"""
\ No newline at end of file
+"""
+
+class SiteUser:
+    """
+    Class to creating site user:
+    name (req): str;
+    email (req): str;
+    access (req): str (admin/moderator/user);
+    logcount (optional): int.
+    """
+    def __init__(self, name, email, access, logcount=0):
+        self.name = name.title()
+        self.email = email
+        self.access = access.lower()
+        self.__logcount = logcount
+
+        # Check if correct role of site user to access
+        if access.lower() not in ['admin', 'moderator', 'user']:
+            raise ValueError("The level of access for user must be 'admin', 'moderator' or 'user'!")
+
+        # Check if email is valid
+        if re.fullmatch(r"\b[A-Za-z0-9._]+@[A-Za-z0-9.-]+\.[A-Z|a-z]{2,7}\b", email):
+            self.email = email
+        else:
+            raise ValueError("Email isn't correct! Please, enter a valid email.")
+
+        # Check if logcount is int
+        if not isinstance(logcount, int):
+            raise ValueError("Count of login must be int!")
+
+    @property
+    def logcount(self):
+        return self.__logcount
+
+    @logcount.setter
+    def logcount(self, logcount: int):
+        """ Use int to set count of login. The count cannot be less than previous value. """
+        if not isinstance(logcount, int):
+            raise ValueError("Count of login must be int!")
+        if logcount > self.__logcount:
+            self.__logcount = logcount
+
+    def __eq__(self, other):
+        """ This function compares two site users by level of access. """
+        if self.access == other.access:
+            return "Users have the same access level."
+        else:
+            return "Users have different access levels."
```

Comment:
не скажу що тут не вірно але сама    ` __eq__` повинна повиртати булеан, тобто більш корректно за умовами задачі буде просто так:
```suggestion
    def __eq__(self, other):
        """ This function compares two site users by level of access. """
        return self.access == other.access
```

Reply:
я спочатку так і зробила, проте хотіла вивісити текст, як написано у таску😄